### PR TITLE
BottomSheet/DialogFragment fixes

### PR DESCRIPTION
**Pull request by bhm**
InfullNavigation, LaunchableBottomSheet and LaunchableIntent got moved to another module.

### CODE REVIEW

**Comment by MasterAccountIFM** [comment]
--- JIRA LINKS ---

File: infullmvp-kotlin/src/main/java/com/infullmobile/android/infullmvp/LaunchableBottomSheet.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.infullmobile.android.infullmvp
+
+import android.support.v4.app.DialogFragment
+import android.support.v4.app.FragmentManager
+
+open class LaunchableBottomSheet(
```
Comment (rainqer) [inline infullmvp-kotlin/src/main/java/com/infullmobile/android/infullmvp/LaunchableBottomSheet.kt]: what's this ?

Comment (bhm) [reply infullmvp-kotlin/src/main/java/com/infullmobile/android/infullmvp/LaunchableBottomSheet.kt]: We have a `LaunchableIntent` class, so we created lately a `LaunchableBottomSheet`. API wrapper to have nice chains for our navigation stack.

Both of these should rather land in a separate artifact.

**Comment by bhm** [comment]
We need to remember to squash this, when merging

**Comment by bhm** [comment]
Pushed the java version

File: infullmvp-basetest/src/main/java/com/infullmobile/android/infullmvp/basetest/InFullMvpBottomSheetDialogFragmentBaseTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @NonNull
+    protected String getString(int stringResourceId) {
+        return RuntimeEnvironment.application.getResources().getString(stringResourceId);
+    }
+
+    @NonNull
+    protected abstract T provideFragment();
+
```
Comment (PawelSwietochowski) [inline infullmvp-basetest/src/main/java/com/infullmobile/android/infullmvp/basetest/InFullMvpBottomSheetDialogFragmentBaseTest.java]: empty space

File: infullmvp/src/main/java/com/infullmobile/android/infullmvp/InFullMvpBottomSheetDialogFragment.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    protected abstract PresenterType getPresenter();
+    protected abstract PresentedViewType getPresentedView();
+    protected abstract void injectIntoGraph();
+
+    @Override
+    public void setupDialog(Dialog dialog, int style) {
+        super.setupDialog(dialog, style);
+        injectIntoGraph();
+        View rootView = View.inflate(getContext(), getPresentedView().getLayoutResId(), null);
```
Comment (PawelSwietochowski) [inline infullmvp/src/main/java/com/infullmobile/android/infullmvp/InFullMvpBottomSheetDialogFragment.java]: final

File: navigation/build.gradle [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    }
+}
+
+dependencies {
+    compile androidAppcompat
+    testCompile 'junit:junit:4.12'
+    compile "org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlin_version"
+}
+
+
```
Comment (PawelSwietochowski) [inline navigation/build.gradle]: empty line

File: navigation/build.gradle [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+ifmmaven {
+    artifactID "navigation"
+    group "com.infullmobile.android.kotlin"
+
+    description "Navigation classes"
+    buildType "release"
+}
+repositories {
```
Comment (PawelSwietochowski) [inline navigation/build.gradle]: And we could add one before this.

File: infullmvp-basetest/src/main/java/com/infullmobile/android/infullmvp/basetest/InFullMvpBottomSheetDialogFragmentBaseTest.java (hunk 2 above)
Comment (bhm) [reply infullmvp-basetest/src/main/java/com/infullmobile/android/infullmvp/basetest/InFullMvpBottomSheetDialogFragmentBaseTest.java]: Done

**Pull request merged**